### PR TITLE
Scans and Sandboxes show "Failed Reason" if set

### PR DESCRIPTION
**Pull request by mrsarm**
Issue: [DN-7002](https://polyswarm.atlassian.net/browse/DN-7002)

API PR: https://github.com/polyswarm/polyswarm-api/pull/245

```diff
$ polyswarm scan url --qrcode-file virus.exe
============================= Artifact Instance =============================
Detections: This scan has failed. Please try again.
Scan id: 81246675384070262
SHA256: None
File type: mimetype: None, extended_info: None
First seen: 2024-08-11 23:03:48 UTC
Last scanned: None
Last seen: None
Status: Failed
+ Failed Reason: Image format not recognized
URL: TrojanDropper_malware_sample
Community: gamma
```

### CODE REVIEW

**Review by sbneto — COMMENTED** [review]
just need to update the api version

File: src/polyswarm/formatters/text.py [hunk 1]
```diff
@@ -108,6 +108,8 @@ def artifact_instance(self, instance, write=True, timeout=False):
         output.extend(self.artifact(instance, write=False))
         if instance.failed:
             output.append(self._red('Status: Failed'))
+            if instance.failed_reason:
+                output.append(self._red(f'Failed Reason: {instance.failed_reason}'))
```
Comment (alanjds) [inline src/polyswarm/formatters/text.py:112]:
```suggestion
                output.append(self._red(f'Failure Reason: {instance.failed_reason}'))
```

File: src/polyswarm/formatters/text.py [hunk 2]
```diff
@@ -362,6 +364,8 @@ def sandbox_task(self, task, write=True):
         output.append(self._white(f'Instance ID: {task.instance_id}'))
         if task.status in ('FAILED', 'FAILED_REIMBURSED', 'TIMEDOUT_REIMBURSED', 'TIMEDOUT'):
             output.append(self._red(f'Status: {task.status}'))
+            if task.status == 'FAILED' and task.artifact.get("failed_reason"):
+                output.append(self._red(f'Failed Reason: {task.artifact.get("failed_reason")}'))
```
Comment (alanjds) [inline src/polyswarm/formatters/text.py:368]:
```suggestion
                output.append(self._red(f'Failure Reason: {task.artifact.get("failed_reason")}'))
```

**Review by alanjds — APPROVED** [review]
May the english wording be improved. I sent a minor suggestion.

**Comment by mrsarm** [comment]
> just need to update the api version

@sbneto is not better just merge with the `dev` update, so development CLI points to a development version of lib? (I make it pointing to API 3.10.0.dev0) so the release to prod / stable is done later when we are sure we have all the stuff we want for both the CLI and API ready for release.

File: src/polyswarm/formatters/text.py (hunk 1 above)
Comment (mrsarm) [reply src/polyswarm/formatters/text.py:112]: Yes I wanted to use the word "Failed" because this goes after the line "Status: Failed", but you are right.

**Review by alanjds — APPROVED** [review]

**Pull request merged**